response_writer.go

```diff
@@ -14,15 +14,6 @@ write json:
 */
 type ResponseWriter struct {
 	Publishing *amqp.Publishing
-	Mandatory  bool
-	Immediate  bool
-}
-
-// NewResponseWriter will create a new response writer with given amqp.Publishing.
-func NewResponseWriter(p *amqp.Publishing) *ResponseWriter {
```

Comment:
No fan of "New" constructors?

Reply:
Perhaps it is usable by unit-tests for packages using amqp-rpc. But we have the ResponseWriter as a stack variable and don't want it to escape to the heap. Although, if we are lucky, the compiler optimizes that.